perception/autoware_image_projection_based_fusion/src/pointpainting_fusion/node.cpp

```diff
@@ -338,12 +347,13 @@ dc   | dc dc dc  dc ||zc|
     p_y = point_camera.y();
     p_z = point_camera.z();
 
-    if (p_z <= 0.0 || p_x > (tan_h_.at(image_id) * p_z) || p_x < (-tan_h_.at(image_id) * p_z)) {
-      continue;
-    }
+    /// check if the point is in the camera view
+    if (p_z <= 0.0) continue;
     // project
     Eigen::Vector2d projected_point = calcRawImageProjectedPoint(
       pinhole_camera_model, cv::Point3d(p_x, p_y, p_z), point_project_to_unrectified_image_);
+    if (projected_point.x() < fov_left_.at(image_id)) continue;
+    if (projected_point.x() > fov_right_.at(image_id)) continue;
```

Comment:
@technolojin Could you change to check fov before `calcRawImageProjectedPoint` function to avoid unnecessary projection calculation for points out of FOV?

Reply:
@badai-nguyen 
We do not know until the lens distortion is corrected. 
In the case the lens warps the view inside, the more points will be inside of FoV. 
In another word, if it comes before the correction, it may miss some points on the edges.

In the case the camera model is the pinhole camera and do not have lens distortion, the previous process `p_x > (tan_h_.at(image_id) * p_z)` will be done in the `calcRawImageProjectedPoint` and it means it doubles the same process anyway.

Reply:
resolved by https://github.com/autowarefoundation/autoware.universe/pull/9637/commits/355c183c4b549dac094a59483f0e2f3169dc5149